src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,324 @@ title: Astro Loader API
+
+## What is a loader?
+
+Astro loaders allow you load data into [content collections](/en/reference/content-collections/), which can then be used in pages and components. The [built-in `glob()` and `file()` loaders](/en/guides/defining-collections/#built-in-loaders) are used to load content from the file system, and you can create your own loaders to load content from other sources. You can define a loader inline in your project's `src/content/config.ts` file, share one loader between multiple collections, or even [publish your loader to NPM as a package](/en/reference/publish-to-npm/) to share with others and be included in our integrations library.
+
+## Loader types
+
+Loaders can be defined either as a simple function that returns an array of entries, or with the more powerful object loader API.
+
+### Inline loaders
+An inline loader is an async function that returns an array or object containing entries. Use this for simple loaders, particularly those that are defined inline in the `src/content/config.ts` file. The function can be async, and must return either an array of entries that each contain a unique `id` field, or an object where each key is a unique ID and each value is the entry. Whenever the loader is invoked, it will clear the store and reload all the entries. 
+
+```ts
+const countries = defineCollection({
+  loader: async () => {
+    const response = await fetch("https://restcountries.com/v3.1/all");
+    const data = await response.json();
+    // Must return an array of entries with an id property
+    // or an object with IDs as keys and entries as values
+    return data.map((country) => ({
+      id: country.cca3,
+      ...country,
+    }));
+  },
+});
+```
+
+### Object loaders
+
+If a loader needs more control over the loading process it can use the object loader API. The loader is defined as an object with a `load()` method that is called at build time to fetch data and update the data store. It allows entries to be updated incrementally, or for the store to be cleared only when necessary. It can also define a schema for the entries, which can be used to validate the data and generate static types. 
+
+The recommended pattern is to define a function that accepts configuration options and returns the loader object, in the same way that you would normally define an Astro integration or Vite plugin. 
+
+
+```ts title=loader.ts
+
+import type { Loader, LoaderContext } from 'astro/loaders';
+
+// Define any options that the loader needs
+export function myLoader(options: { url: string }): Loader {
+  // Configure the loader 
+  const feedUrl = new URL(options.url);
+  // Return a loader object
+  return {
+    name: "my-loader";
+    // Called when updating the collection.
+    load: async (context: LoaderContext): Promise<void> => {
+      // Load data and update the store
+    }
+    // Optionally, define the schema of an entry. 
+    // It will be overridden by user-defined schema. 
```

Comment:
I might have missed it, but where would a user define a schema?

Reply:
as a property of defineCollection

Reply:
Ooh, right. The loaders are used inside the definition, not replacing it. That is what I got wrong.